From cce4b47f4453b94727b8bb7457b908517e200d9f Mon Sep 17 00:00:00 2001
From: Phat Pham <phat.pham.zg@rvc.renesas.com>
Date: Fri, 8 Mar 2019 14:28:44 +0700
Subject: [PATCH 779/909] arm64: dts: r8a7795: Connect SDHI to IPMMU-DS1

commit 4bd36fe168284318c1091e998e7aa8351ff34528 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

Add IPMMU-DS1 to the SDHI device node.

Signed-off-by: Phat Pham <phat.pham.zg@rvc.renesas.com>
Signed-off-by: Hai Nguyen Pham <hai.pham.ud@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/renesas/r8a7795-es1.dtsi | 16 ++++++++++++++++
 arch/arm64/boot/dts/renesas/r8a7795.dtsi     |  4 ++++
 2 files changed, 20 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a7795-es1.dtsi b/arch/arm64/boot/dts/renesas/r8a7795-es1.dtsi
index b10d176e1a26..f14579650f8a 100644
--- a/arch/arm64/boot/dts/renesas/r8a7795-es1.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a7795-es1.dtsi
@@ -273,6 +273,22 @@
 	renesas,ipmmu-main = <&ipmmu_mm 7>;
 };
 
+&sdhi0 {
+	/delete-property/ iommus;
+};
+
+&sdhi1 {
+	/delete-property/ iommus;
+};
+
+&sdhi2 {
+	/delete-property/ iommus;
+};
+
+&sdhi3 {
+	/delete-property/ iommus;
+};
+
 &du {
 	vsps = <&vspd0 &vspd1 &vspd2 &vspd3>;
 };
diff --git a/arch/arm64/boot/dts/renesas/r8a7795.dtsi b/arch/arm64/boot/dts/renesas/r8a7795.dtsi
index b024a3b67f32..94ddbcbd081c 100644
--- a/arch/arm64/boot/dts/renesas/r8a7795.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a7795.dtsi
@@ -2806,6 +2806,7 @@
 			max-frequency = <200000000>;
 			power-domains = <&sysc R8A7795_PD_ALWAYS_ON>;
 			resets = <&cpg 314>;
+			iommus = <&ipmmu_ds1 32>;
 			status = "disabled";
 		};
 
@@ -2818,6 +2819,7 @@
 			max-frequency = <200000000>;
 			power-domains = <&sysc R8A7795_PD_ALWAYS_ON>;
 			resets = <&cpg 313>;
+			iommus = <&ipmmu_ds1 33>;
 			status = "disabled";
 		};
 
@@ -2830,6 +2832,7 @@
 			max-frequency = <200000000>;
 			power-domains = <&sysc R8A7795_PD_ALWAYS_ON>;
 			resets = <&cpg 312>;
+			iommus = <&ipmmu_ds1 34>;
 			status = "disabled";
 		};
 
@@ -2842,6 +2845,7 @@
 			max-frequency = <200000000>;
 			power-domains = <&sysc R8A7795_PD_ALWAYS_ON>;
 			resets = <&cpg 311>;
+			iommus = <&ipmmu_ds1 35>;
 			status = "disabled";
 		};
 
-- 
2.17.1

